TYPO3.Media/Classes/TYPO3/Media/Domain/Service/AssetService.php

```diff
@@ -37,66 +38,47 @@ class AssetService
      *
      * @param AssetInterface $asset
      * @param ThumbnailConfiguration $configuration
-     * @return array with keys "width", "height" and "src"
+     * @return array|null Array with keys "width", "height" and "src" if the thumbnail generation work or null
      */
     public function getThumbnailUriAndSizeForAsset(AssetInterface $asset, ThumbnailConfiguration $configuration)
     {
-        if ($asset instanceof ImageInterface) {
-            $thumbnailImage = $this->thumbnailService->getThumbnail($asset, $configuration);
+        $thumbnailImage = $this->getImageThumbnail($asset, $configuration);
+        if ($thumbnailImage instanceof ImageInterface) {
+            if ($thumbnailImage instanceof Thumbnail && $thumbnailImage->isTransient()) {
```

Comment:
it would return a resource or a string, how do you see a unified method helping here?